compute_sdk/globus_compute_sdk/sdk/login_manager/manager.py

```diff
@@ -19,10 +19,13 @@
 
 
 def _get_funcx_all_scope() -> str:
-    return os.getenv(
+    scope = get_env_var_with_deprecation(
+        "GLOBUS_COMPUTE_SCOPE",
         "FUNCX_SCOPE",
         "https://auth.globus.org/scopes/facd7ccc-c5f4-42aa-916b-a0e270e2c2a9/all",
     )
+    # mypy doesn't realize that the default means a guaranteed str
+    return t.cast(str, scope)
```

Comment:
Another way of educating mypy is with an assert, for cases like this:
```python
scope = mypy_questioned_type
assert isinstance(scope, str)
return scope
```
Pick your poison.

The advantage of this route is that for anyone who choose to run optimized (`python -O`), then all asserts are removed from the bytecode:

```console
-O     Remove  assert  statements  and  any code conditional on the value
       of __debug__; augment the filename for compiled (bytecode) files by
       adding .opt-1 before the .pyc extension.
```